src/pyfmi/fmi3.pyx

```diff
@@ -465,6 +531,70 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
             Calls the low-level FMI function: fmi3SetTime or fmi3GetTime.
     """)
 
+    def get_variable_declared_type(self, variable_name):
+        """
+        Return the given variables declared type.
+
+        Parameters::
+
+            variable_name --
+                The name of the variable.
+
+        Returns::
+
+            The declared type.
+        """
+        cdef int item_value
+        cdef unsigned int enum_size
+        cdef FMIL3.fmi3_base_type_enu_t basetype
+        cdef FMIL3.fmi3_import_variable_t* variable
+        cdef FMIL3.fmi3_import_variable_typedef_t* variable_type
+        cdef FMIL3.fmi3_import_enumeration_typedef_t * enumeration_type
+        cdef FMIL3.fmi3_string_t type_name
+        cdef FMIL3.fmi3_string_t type_desc
+        cdef FMIL3.fmi3_string_t type_quantity
+        cdef FMIL3.fmi3_string_t item_desc
+        cdef FMIL3.fmi3_string_t item_name
+
+        variable_name = pyfmi_util.encode(variable_name)
+        cdef char* variablename = variable_name
+
+        variable = FMIL3.fmi3_import_get_variable_by_name(self._fmu, variablename)
+        if variable == NULL:
+            raise FMUException("The variable %s could not be found."%variablename)
```

Comment:
By having `variable_name = pyfmi_util.encode(variable_name)`, `variable_name` becomes a `byte` string, which makes the string formatting a bit messy here. 

I recommend to instead use 

```
variable_name_bytes = pyfmi_util.encode(variable_name)
cdef char* variablename = variable_name_bytes
```
and do the string formatting of both Exception messages using `variable_name` instead.